cpp/dolfin/io/XDMFFile.cpp

```diff
@@ -1341,11 +1427,28 @@ void XDMFFile::add_function(MPI_Comm mpi_comm, pugi::xml_node& xml_node,
 
   // Get all local data
   const la::PETScVector& u_vector = *u.vector();
-  std::vector<double> local_data;
+  std::vector<PetscScalar> local_data;
   u_vector.get_local(local_data);
 
+#ifdef PETSC_USE_COMPLEX
+  // FIXME: Avoid copies by writing directly a compound data
```

Comment:
Note: With fixing this, this can probably also can be done with
```
#ifdef PETSC_USE_COMPLEX
  std::vector<std::string> components { ... }
#else
  std::vector<std::string> components { ... }
#endif.
```